### ISSUE TITLE
How to work with large binary files?

### ISSUE THREAD

**Issue by roygarbriceno** [simple-odata-client/Simple.OData.Client#10]
So far I found this client a lot better (and easier) than the WCF Data Services, however I don't know how to work with binary files, I read in MSDN about creating a stream provider for named resources but that can be used with this library? or are there other alternatives o samples?

Thanks in advance

**Comment by elbecita** [comment]
Hi!
We're using the library to test some OData endpoints that accept images. To do it, we modify the content of the request message in the BeforeRequest of the ODataClientSettings. This way me managed to send the stream to the endpoint. Maybe it can help to anybody that wants to send a byte stream to an endpoint. ;)

**Comment by object** [comment]
Thank you for sharing your experience. Actually I started working on streaming support that should be implemented in version 5.0, to be released sometime this fall.

**Comment by mbrennan2015** [comment]
Vagif, any update on blob streaming support in simple.odata?  Is there a prerelease available?

**Comment by ErikWitkowski** [comment]
Any plan for a prerelease? Really need stream support.

**Comment by object** [comment]
Yes, there is in fact a pre-release. If you fetch master branch of this project you can have a preview of the version 5 (check out its integration tests showing how to deal with streams).

**Comment by ErikWitkowski** [comment]
Thank you! I already checked the code on master branch, was wondering if it's already pre-release material.
